### PR TITLE
#37: Refactor in TypeScript! (closes #37)

### LINKED ISSUE #37: Refactor in TypeScript!
## requirements
Refactor in TypeScript!

## specs
Refactor in TypeScript!

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Closes #37

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.}

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,36 +36,46 @@
...
   "dependencies": {
+    "@types/classnames": "2.2.3",
+    "@types/prop-types": "15.5.2",
+    "@types/react": "16.0.25",
     "browser-cookie-lite": "1.0.4",
-    "classnames": "^2.1.5",
-    "lodash.omit": "^3.1.0",
-    "tcomb-react": "^0.9.3"
+    "classnames": "2.2.3",
+    "lodash.omit": "^3.1.0"
```
Comment (giogonzo) [inline package.json]: no `@types` for `omit`?

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          {
+            loader: 'babel-loader',
+            options: {
+              presets: [['buildo', { env: 'react' }]]
+            }
+          },
+          {
+            loader: 'ts-loader',
+            options: {
+              transpileOnly: true,
```
Comment (giogonzo) [inline webpack.config.js]: this + the fact that there's no typescript build run on the CI makes TS quite useless..

File: package.json (hunk 1 above)
Comment (giogonzo) [reply package.json]: oh I see you have declared it as `any` manually 😢

File: webpack.config.js (hunk 2 above)
Comment (FrancescoCioria) [reply webpack.config.js]: you're right, we should add a script `typecheck` that runs `tsc --noEmit` and call it in the CI!

File: .drone.yml [hunk 3]
```diff
@@ -1,36 +1,19 @@
...
-
-notify:
-  downstream:
-    repositories:
-      - buildo/react-components
-    token: $$NOTIFY_BRC_TOKEN
-    fork: true
-    when:
-      branch: master
+# notify:
```
Comment (gabro) [inline .drone.yml]: why is this commented?

**Pull request merged**